### DISCUSSION TITLE
MIP-3: Sized Collection Feature Deprecation

### DISCUSSION THREAD

**Discussion by blockiosaurus** [metaplex-foundation/mip#6] [Announcements]
Discussions for [MIP-3: Sized Collection Feature Deprecation](https://github.com/metaplex-foundation/mip/blob/main/mip-3.md)

**Comment by SebastianBor** [comment]
The ability to have on-chain collection size is critical for any NFT based governance where the size of the collection is dynamic. If the collection is used to control assets like DAO treasury then it's also critical for it to be accurate because otherwise it defeats the purpose.
- If the on-chain collection size is lower than the actual size then it gives disproportionally more governance power to individual NFT holders 
- If the on-chain collection size is larger than the actual size then there is a risk the DAO could be bricked because it'll be impossible to ever reach a quorum 

if we can't get the correct on-chain collection size with the existing solution then there is no need to keep it. However we would need a solution for this and if pNFTs can solve the problem then that works.

**Comment by statikdev** [comment]
>## `CollectionDetails` Replacement Options
>
> In the case of Sized Collections deprecation, there are several possible options for changes that can be made to the data structure in the on-chain metadata. These options are listed out with an explanation and pros and cons below.
>
>**Removal** - This would entail completely removing the struct to reclaim space, but would have negative consequences of no longer being able to easily tell if an NFT is a collection NFT.
>
>**Leave it alone** - In this case the field would be completely untouched, but this has the potential to confuse new users by having a zero value for Collection Size.
>
>**Empty Struct (Recommended)** - The struct could be left in place but the collection size field could be removed. This would reclaim eight bytes of data and leave the ability to detect if the NFT is a collection NFT. This also leaves in the possibility of adding more details to the Collection Details struct in the future.

I vote to leave an empty struct. Being able to determine if it is a collection NFT is helpful from an aggregation standpoint. Without the struct, collection NFTs minted will look like any other NFT when they have the explicit use of being a collection NFT. The easiest option from there is to track if an NFT address is used as a verified collection key for other NFTs, complicating the logic.
